src/plugins/intel_gpu/src/kernel_selector/kernels/permute/permute_kernel_bfzyx_to_bfyxz.cpp

```diff
@@ -0,0 +1,184 @@
+    k.EnableInputDataType(Datatype::INT8);
+    k.EnableInputDataType(Datatype::UINT8);
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::INT8);
+    k.EnableOutputDataType(Datatype::UINT8);
+    k.EnableOutputDataType(Datatype::INT32);
+    k.EnableOutputDataType(Datatype::INT64);
+    k.EnableDifferentTypes();
+    k.EnableInputLayout(DataLayout::bfzyx);
+    k.EnableOutputLayout(DataLayout::bfzyx);
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    return k;
+}
+
+static inline bool isMultipleDefaultTileSize(const size_t size) {
+    return size % DEFAULT_TILE_SIZE == 0;
+}
+
+static inline size_t GetTileSize(const permute_params& params) {
+    // supports 4x4 or 8x8 tiling
+    if (!isMultipleDefaultTileSize(params.inputs[0].X().v) || !isMultipleDefaultTileSize(params.inputs[0].Z().v))
+        return MIN_TILE_SIZE;
+
+    if ((params.inputs[0].GetDType() == Datatype::INT64) || (params.outputs[0].GetDType() == Datatype::INT64))
+        return MIN_TILE_SIZE;
+
+    return DEFAULT_TILE_SIZE;
+}
+
+
+static inline std::string GetTiledOutputOrder() {
+    return  "b, f, y, (x * TILE_SIZE + lh), (z * TILE_SIZE)";
+}
+
+static inline std::string GetTiledInputOrder() {
+    return  "b, f, (z * TILE_SIZE + lh), y, (x * TILE_SIZE)";
+}
+
+JitConstants PermuteKernel_bfzyx_to_bfyxz::GetJitConstants(const permute_params& params, const CommonDispatchData& dispatchData) const {
+    auto jit = Parent::GetJitConstants(params, dispatchData);
+    size_t tile_size = GetTileSize(params);
+    size_t vector_width = tile_size;
+    uint64_t total_lws = dispatchData.lws[0] * dispatchData.lws[1] * dispatchData.lws[2];
+    jit.AddConstant(MakeJitConstant("VEC_WIDTH", vector_width));
+    jit.AddConstant(MakeJitConstant("INPUT0_TILED_ORDER", GetTiledInputOrder()));
+    jit.AddConstant(MakeJitConstant("OUTPUT_TILED_ORDER", GetTiledOutputOrder()));
+    jit.AddConstant(MakeJitConstant("TILE_SIZE", tile_size));
+    jit.AddConstant(MakeJitConstant("NZ_TILES", CeilDiv(params.inputs[0].Z().v, tile_size)));
+
+    std::string normal_tile_cond = "true";
+    std::string x_remainder_cond = "true";
+    std::string f_remainder_cond = "true";
+
+    if (params.inputs[0].X().v % tile_size) {
+        jit.AddConstant(MakeJitConstant("X_REMAINDER_ITEM", params.inputs[0].X().v / tile_size));
+        jit.AddConstant(MakeJitConstant("X_REMAINDER_SIZE", params.inputs[0].X().v % tile_size));
+        jit.AddConstant(MakeJitConstant("X_REMAINDER_SIZE_AS_VECTOR", CeilDiv(params.inputs[0].X().v % tile_size, vector_width)));
+        normal_tile_cond += " && (x < X_REMAINDER_ITEM)";
+        x_remainder_cond += " && (x == X_REMAINDER_ITEM)";
+        f_remainder_cond += " && (x < X_REMAINDER_ITEM)";
+    }
+    if (params.inputs[0].Z().v % tile_size) {
```

Comment:
This kernel was tuned for permuting bfzyx ->  bfxzy {1, 3, 85, 20, 20} shapes (or similar) from yolov model.  [85 * 20, 20] to [20, 85 * 20]
I mean that here with tile_size equal 4 it will be a reminder for Z axes (85 % 4 != 0) but if unite z and y axes,   [85 * 20, 20] to [20, 85 * 20], then will not be any reminder (85*20 % 4 ==0), then everything works under normal conditions.

Reply:
this kernel supports any shapes, including shapes from yolo model.